engine/src/cmd/drawable.cpp

```diff
@@ -0,0 +1,247 @@
+#include "drawable.h"
+#include "vsfilesystem.h"
+#include "gfx/mesh.h"
+#include "gfx/quaternion.h"
+#include "unit_generic.h"
+
+#include <boost/algorithm/string.hpp>
+
+
+Drawable::Drawable() :
+                   animatedMesh(true),
+                   activeAnimation(0),
+                   timeperframe(3.0),
+                   done(true),
+                   activeMesh(0),
+                   nextactiveMesh(1),
+                   infiniteLoop(true),
+                   loopCount(0),
+                   curtime(0.0)
+
+{}
+
+bool Drawable::DrawableInit(const char *filename, int faction,
+        Flightgroup *flightgrp, const char *animationExt)
+{
+    string fnam(filename);
+    string::size_type pos = fnam.find('.');
+    string anifilename = fnam.substr(0, pos);
+
+    if(animationExt)
+        anifilename += string("_") + string(animationExt);
+
+	std::vector< Mesh* > *meshes = new vector<Mesh *>();
+	int i = 1;
+	char count[30] = "1";
+	string dir = anifilename;
+	while(true)
+	{
+		sprintf( count, "%d", i );
+		string unit_name = anifilename;
+		boost::algorithm::to_lower(unit_name); //toLowerCase(anifilename) + "_";
+
+		if(i < 10)
+			unit_name += "0";
+		if(i < 100)
+			unit_name += "0";
+		if(i < 1000)
+			unit_name += "0";
+		if(i < 10000)
+			unit_name += "0";
+		if(i < 100000)
+			unit_name += "0";
+
+		unit_name += count;
+		string path = dir + "/" + unit_name + ".bfxm";
+		if( VSFileSystem::FileExistsData( path, VSFileSystem::MeshFile ) != -1 )
+		{
+			Mesh *m = Mesh::LoadMesh( path.c_str(), Vector(1,1,1), faction, flightgrp );
+			meshes->push_back( m );
+	#ifdef DEBUG_MESH_ANI
+            // TODO: search and replace cerr with BOOST_LOG_TRIVIAL(error)
+			cerr << "Animated Mesh: " << path << " loaded - with: " << m->getVertexList()->GetNumVertices() << " vertices." << endl;
+	#endif
+		}
+		else
+			break;
+		i++;
+	}
+
+
+	if( meshes->size() != 0 )
+	{
+		//FIXME: an animation is created only for the first submesh
+		string animationName;
+		sprintf( count, "%lu", (unsigned long)meshes->size() );
+		if(!animationExt)
+			animationName = string(count); //if there is no extension given, the animations are called by their load order, 1, 2 ,3 ....10..
+		else
+			animationName = animationExt;
+		addAnimation(meshes, animationName.c_str());
```

Comment:
even though it's only 1 line, we really should have the if blocks captured in braces to avoid accidental bugs.